File: MCP_GATEWAY/SKill_SOCKET-Mcp_gateway/agents/RoadmapAgent.js
class RoadmapAgent {
    constructor(llmClient, toolRegistry) {
        this.llmClient=llmClient;
        this.websearchTool=toolRegistry.get('web_search');
    }

    async run(topic){
        console.log(`Roadmap Agent started for topic: "${topic}"`);
        const searchResults = await this.websearchTool(`learning path and key concepts for ${topic}`);
        const context = searchResults.results.map(res => `Source: ${res.content}`).join('\n\n');
        const prompt = `Topic: "Learn ${topic}"\n\nContext:\n${context}\n\nBased on the context, generate a detailed, step-by-step learning roadmap in Markdown. Include stages (Beginner, Intermediate, Advanced) with key concepts and project ideas.`;
        const roadmap = await this.llmClient.generateText(prompt, 0.7);
        return { roadmap };
}
}
module.exports=RoadmapAgent;

File: MCP_GATEWAY/SKill_SOCKET-Mcp_gateway/agents/roadmap_agent.py
class RoadmapAgent:
    """Generates a step-by-step learning roadmap for any topic."""

    def __init__(self, llm_client, tool_registry):
        self.llm_client = llm_client
        self.websearch_tool = tool_registry.get("web_search")

    async def run(self, topic: str) -> dict:
        print(f'Roadmap Agent started for topic: "{topic}"')

        search_results = await self.websearch_tool(
            f"learning path and key concepts for {topic}"
        )
        results = search_results.get("results", [])
        context = "\n\n".join(f"Source: {r['content']}" for r in results)

        prompt = (
            f'Topic: "Learn {topic}"\n\n'
            f"Context:\n{context}\n\n"
            "Based on the context, generate a detailed, step-by-step learning roadmap "
            "in Markdown. Include stages (Beginner, Intermediate, Advanced) with key "
            "concepts and project ideas."
        )

        roadmap = await self.llm_client.generate_text(prompt, 0.7)
        return {"roadmap": roadmap}
